engine/transaction/record/record_outline_test.go

```diff
@@ -278,46 +264,53 @@ func TestRecordOutlineOpReturnErrorCases(t *testing.T) {
 			},
 			expectErr: txerrors.ErrAnnotationMismatch,
 		},
-		"error during broadcasting": {
-			broadcaster: newMockBroadcaster().withError(errors.New("broadcast error")),
-			repo: newMockRepository().withOutput(database.Output{
-				TxID: givenTXWithOpReturn(t).InputUTXO(0).TxID,
-				Vout: givenTXWithOpReturn(t).InputUTXO(0).Vout,
-			}),
-			outline: &outlines.Transaction{
-				BEEF: givenTXWithOpReturn(t).BEEF(),
-				Annotations: transaction.Annotations{
-					Outputs: transaction.OutputsAnnotations{
-						0: &transaction.OutputAnnotation{
-							Bucket: bucket.Data,
-						},
-					},
-				},
-			},
-			expectErr: txerrors.ErrTxBroadcast,
-		},
 	}
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
 			// given:
-			service := record.NewService(tester.Logger(t), test.repo, test.broadcaster)
-			initialOutputs := test.repo.getAllOutputs()
-			initialData := test.repo.getAllData()
+			given, then := testabilities.New(t)
+			service := given.WithStoredOutputs(test.storedOutputs...).NewRecordService()
 
 			// when:
 			err := service.RecordTransactionOutline(context.Background(), test.outline)
 
 			// then:
-			require.Error(t, err)
-			require.ErrorIs(t, err, test.expectErr)
-
-			// ensure that no changes were made to the repository
-			require.ElementsMatch(t, initialOutputs, test.repo.getAllOutputs())
-			require.ElementsMatch(t, initialData, test.repo.getAllData())
+			then.WithErrorIs(err, test.expectErr).NothingChanged()
 		})
 	}
 }
 
+func TestOnBroadcastErr(t *testing.T) {
```

Comment:
What do you think about more complex test cases like multiple error handling?
**Example / pseudo code**:
```go
func TestRecordOutlineMultipleFailures(t *testing.T) {
    // given:
    given, then := testabilities.New(t)
    service := given.
        WithStoredOutputs(database.Output{
            TxID:       givenTXWithOpReturn(t).InputUTXO(0).TxID,
            Vout:       givenTXWithOpReturn(t).InputUTXO(0).Vout,
            SpendingTX: nil,
        }).
        WillFailOnSaveTX(errors.New("repository save error")).
        WillFailOnBroadcast(errors.New("broadcaster error")).
        NewRecordService()

    // and:
    outline := &outlines.Transaction{
        BEEF: givenTXWithOpReturn(t).BEEF(),
        Annotations: transaction.Annotations{
            Outputs: transaction.OutputsAnnotations{
                0: &transaction.OutputAnnotation{
                    Bucket: bucket.Data,
                },
            },
        },
    }

    // when:
    err := service.RecordTransactionOutline(context.Background(), outline)

    // then:
    then.WithErrorIs(err, txerrors.ErrTxRepositorySave).NothingChanged()
    then.WithErrorIs(err, txerrors.ErrTxBroadcast).NothingChanged()
}
```

Reply:
That won't work like you're showing.
The error can be either `ErrTxBroadcast` or `ErrTxRepositorySave` - but not both at the same time.
Taking this into account - I don't think, such test will be beneficial.

Reply:
Ok, undesrtand 👍